gcp_variant_transforms/beam_io/vcf_header_io.py

```diff
@@ -117,16 +171,39 @@ def _merge_header_fields(self, source, to_merge):
     for key, to_merge_value in to_merge.iteritems():
       if key not in source:
         source[key] = to_merge_value
-      else:
-        source_value = source[key]
-        # Verify num and type fields are the same for infos and formats
-        if (source_value.keys() != to_merge_value.keys()
-            or ('num' in source_value and 'type' in source_value
-                and (source_value['num'] != to_merge_value['num']
-                     or source_value['type'] != to_merge_value['type']))):
-          raise ValueError(
-              'Incompatible number or types in header fields: {}, {}'.format(
-                  source_value, to_merge_value))
+        continue
+      source_value = source[key]
+      if source_value.keys() != to_merge_value.keys():
+        raise ValueError('Incompatible header fields: {}, {}'.format(
+            source_value, to_merge_value))
+      resolver = HeaderConflictResolver()
+      merged_value = OrderedDict()
+      for source_field_key, source_field_value in source_value.iteritems():
+        # We only care about mistmach in 'num' and 'type' fields.
+        if (to_merge_value[source_field_key] == source_field_value or
+            source_field_key not in ['num', 'type']):
+          merged_value.update({source_field_key : source_field_value})
+          continue
+        # There is a conflict in header fields. Try to resolve it.
+        can_resolve_conflict = resolver.can_resolve(
+            source_field_value, to_merge_value[source_field_key])
+        if force_merge_conflicts and can_resolve_conflict:
+          resolution_field_value = resolver.resolve(
+              source_field_value, to_merge_value[source_field_key])
+          merged_value.update({source_field_key : resolution_field_value})
+        elif not force_merge_conflicts and can_resolve_conflict:
```

Comment:
This statement can become `elif resolver.can_resolve(source_field_value, to_merge_value[source_field_key])`.

Reply:
Done!